res/controllers/Traktor-Kontrol-S2-MK3-hid-scripts.js

```diff
@@ -0,0 +1,978 @@
+        var sampler = TraktorS2MK3.samplerHotcuesRelation[field.group][hotcueNumber];
+        if (TraktorS2MK3.shiftPressed[field.group]) {
+            var playing = engine.getValue("[Sampler" + sampler + "]", "play_indicator");
+            if (playing) {
+                engine.setValue("[Sampler" + sampler + "]", "cue_default", field.value);
+            } else {
+                engine.setValue("[Sampler" + sampler + "]", "eject", field.value);
+                // Reset eject button
+                engine.setValue("[Sampler" + sampler + "]", "eject", !field.value);
+            }
+        } else {
+            var loaded = engine.getValue("[Sampler" + sampler + "]", "track_loaded");
+            if (loaded) {
+                engine.setValue("[Sampler" + sampler + "]", "cue_gotoandplay", field.value);
+            } else {
+                engine.setValue("[Sampler" + sampler + "]", "LoadSelectedTrack", field.value);
+            }
+        }
+    }
+};
+
+TraktorS2MK3.headphoneHandler = function (field) {
+    if (field.value === 0) {
+        return;
+    }
+
+    script.toggleControl(field.group, "pfl");
+};
+
+TraktorS2MK3.selectTrackHandler = function (field) {
+    var delta = 1;
+    if ((field.value + 1) % 16 === TraktorS2MK3.browseState[field.group]) {
+        delta = -1;
+    }
+
+    if (TraktorS2MK3.shiftPressed[field.group]) {
+        engine.setValue("[Library]", "MoveHorizontal", delta);
+    } else {
+        engine.setValue("[Library]", "MoveVertical", delta);
+    }
+
+    TraktorS2MK3.browseState[field.group] = field.value;
+};
+
+TraktorS2MK3.loadTrackHandler = function (field) {
+    if (field.value === 0) {
+        return;
+    }
+
+    if (TraktorS2MK3.shiftPressed[field.group]) {
+        engine.setValue(field.group, "eject", field.value);
+        // Reset eject button
+        engine.setValue(field.group, "eject", 0);
+    } else {
+        engine.setValue(field.group, "LoadSelectedTrack", field.value);
+    }
+};
+
+TraktorS2MK3.addTrackHandler = function (field) {
+    TraktorS2MK3.outputHandler(field.value, field.group, "addTrack");
+
+    if (field.value === 0) {
+        return;
+    }
+
+    if (TraktorS2MK3.shiftPressed[field.group]) {
+        engine.setValue("[Library]", "AutoDjAddTop", field.value);
+    } else {
+        engine.setValue("[Library]", "AutoDjAddBottom", field.value);
+    }
+};
+
+TraktorS2MK3.maximizeLibraryHandler = function (field) {
+    if (field.value === 0) {
+        return;
+    }
+
+    script.toggleControl("[Master]", "maximize_library");
+};
+
+TraktorS2MK3.selectLoopHandler = function (field) {
+    if ((field.value + 1) % 16 === TraktorS2MK3.loopSizeState[field.group]) {
+        engine.setValue(field.group, "loop_halve", 1);
+    } else {
+        engine.setValue(field.group, "loop_double", 1);
+    }
+    TraktorS2MK3.loopSizeState[field.group] = field.value;
+};
+
+TraktorS2MK3.activateLoopHandler = function (field) {
+    if (field.value === 0) {
```

Comment:
ping

Reply:
Fixed this in https://github.com/mi01/mixxx/commit/3f0fd915ae129993968be79bcb867eb8f22bb468

But when looking on this function again i noticed, that the current mapping is maybe not so good. There is only one button on the controller regarding loops and mapping only `reloop_toggle` and `reloop_andstop` is not usefull. I thought to distinguish here, if a loop is currently enabled or not. If yes, than this button maps `reloop_toggle` and `reloop_andstop`, otherwhise `beatloop_activate` and `beatlooproll_activate`. What do you think?

Reply:
Yes, this could be mapped better. It looks like you haven't mapped anything for pressing the other encoder, so consider that as well. It is useful to have access to `reloop_toggle` when there is no loop active though. How about:

press loop size encoder: beatloop_activate
shift + press loop size encoder: reloop_toggle

press beatjump encoder: beatlooproll_activate
shift + press beatjump encoder: reloop_andstop

Reply:
I like your proposal and updated the script like this in https://github.com/mi01/mixxx/commit/86effbbcbd4faed8f214f4bd66d5430d4422a349. I will update the Wiki page now.